api/src/main/java/org/apache/iceberg/metrics/MetricsContext.java

```diff
@@ -42,6 +42,20 @@ enum Unit {
     public String displayName() {
       return displayName;
     }
+
+    public static Unit fromDisplayName(String displayName) {
```

Comment:
We would normally reuse `valueOf` to make this simpler and not require updating if more symbols are added:

```java
return Unit.valueOf(dispayName.toUpperCase(Locale.ROOT));
```

Reply:
done